smaug/python/graph.py

```diff
@@ -106,17 +106,35 @@ def add_node(
 
     return output_tensors
 
-  def get_node(self, node_name):
-    """Return a node in the graph proto by its name."""
+  def get_node(self, node_name, recursive=False):
+    """Return a node in the graph proto by its name.
+
+    Args:
+      node_name: Node name.
+      recursive: If true, recursively search the node in the parent graphs.
+
+    Returns:
+      A NodeProto if we find the node.
+    """
     for i in range(len(self.graph.nodes)):
       if self.graph.nodes[i].name == node_name:
         return self.graph.nodes[i]
-    return None
+    if recursive and self._parent_graph is not None:
+      return self._parent_graph.get_node(node_name, True)
 
   def get_nodes(self):
     """Return nodes in the graph proto."""
     return self.graph.nodes
 
+  def get_root_graph(self):
+    """Return the root graph."""
+    root = self
+    while True:
+      if root._parent_graph is None:
+        break
+      root = root._parent_graph
+    return root
+
   def create_unique_name(self, name, mark_as_used=True):
```

Comment:
Do we still need `mark_as_used` , or `create_unique_name` to be a public API?

Reply:
Removed `mark_as_used` as we don't need it anymore. `create_unique_name` needs to be a public API as we still have use cases for that.